In [1]:
from functools import partial
import json
import torch
from transformers import AutoModelForCausalLM, Trainer, TrainingArguments, DataCollatorForSeq2Seq
from datasets import Dataset
from swanlab.integration.transformers import SwanLabCallback
import swanlab
from modelscope import AutoTokenizer

from peft import LoraConfig, get_peft_model, TaskType

import torch
import random
import numpy as np
import os

seed = 7
torch.manual_seed(seed)
torch.cuda.manual_seed(seed)
torch.cuda.manual_seed_all(seed)
random.seed(seed)
np.random.seed(seed)
os.environ['PYTHONHASHSEED'] = str(seed)

g:\software\anaconda\envs\pytorch\Lib\site-packages\torch\cuda\__init__.py:65: FutureWarning: The pynvml package is deprecated. Please install nvidia-ml-py instead. If you did not install pynvml directly, please report this to the maintainers of the package that installed pynvml for you.
  import pynvml  # type: ignore[import]


In [8]:
model_id = "qwen2.5-1.5B-Instruct"  
model_path = "G:/model_weights/models/Qwen/qwen2.5-1.5B-Instruct"
data_path = "D:/doc/my_project/lianxi/home_work/stage_3/data_set"

# 1.数据预处理

In [3]:
class TCMNERDataset():
    def __init__(self, data_path):
        self.label_set = set()
        self.data = self.load_data(data_path)
        # self.dataset = Dataset.from_list(self.data)
        self.id2label = {idx: item for idx, item in enumerate(self.label_set)}
        self.label2id = {item: idx for idx, item in self.id2label.items()}


    def load_data(self, data_path):
        dataset = []
        with open(data_path, 'r', encoding='utf-8') as f:
            trunk = f.read().split('\n\n')
            for line in trunk:
                input, output = '', []
                for idx, string in enumerate(line.split('\n')):
                    if not string.strip():  # 跳过空行
                        continue
                    text, tag = string.split(' ')  # 假设文本和标签之间用空格分隔
                    input += text
                    if tag.startswith('B-'):
                        self.label_set.add(tag[2:])
                        output.append([text, tag[2:]])  # 添加标签内容
                    elif tag.startswith('I-') and output:
                        output[-1][0] += text  # 更新标签内容
                dataset.append({'input': input, 'output': json.dumps(output, ensure_ascii=False)})  # 将标签列表转换为JSON字符串
        return dataset

In [4]:
from pathlib import Path
train_data_path = Path(data_path) / 'medical.train'
test_data_path = Path(data_path) / 'medical.test'
dev_data_path = Path(data_path) / 'medical.dev'

In [5]:
train_data = TCMNERDataset(train_data_path)
test_data = TCMNERDataset(test_data_path)
dev_data = TCMNERDataset(dev_data_path)

In [9]:
tokenizer = AutoTokenizer.from_pretrained(model_path)
model = AutoModelForCausalLM.from_pretrained(model_path)#, id2label=train_data.id2label, label2id=train_data.label2id)
model.enable_input_require_grads()  # 开启梯度检查点时，要执行该方法

In [7]:
train_data.data[0], train_data.id2label, train_data.label2id

({'input': '现头昏口苦', 'output': '[["口苦", "临床表现"]]'},
 {0: 'B-方剂',
  1: 'B-中医治则',
  2: 'I-中药',
  3: 'I-中医治疗',
  4: 'I-方剂',
  5: 'I-西医诊断',
  6: 'B-中药',
  7: 'B-临床表现',
  8: 'I-中医治则',
  9: 'B-其他治疗',
  10: 'I-中医证候',
  11: 'B-西医诊断',
  12: 'I-其他治疗',
  13: 'B-中医诊断',
  14: 'B-西医治疗',
  15: 'I-西医治疗',
  16: 'I-中医诊断',
  17: 'I-临床表现',
  18: 'B-中医证候',
  19: 'B-中医治疗'},
 {'B-方剂': 0,
  'B-中医治则': 1,
  'I-中药': 2,
  'I-中医治疗': 3,
  'I-方剂': 4,
  'I-西医诊断': 5,
  'B-中药': 6,
  'B-临床表现': 7,
  'I-中医治则': 8,
  'B-其他治疗': 9,
  'I-中医证候': 10,
  'B-西医诊断': 11,
  'I-其他治疗': 12,
  'B-中医诊断': 13,
  'B-西医治疗': 14,
  'I-西医治疗': 15,
  'I-中医诊断': 16,
  'I-临床表现': 17,
  'B-中医证候': 18,
  'B-中医治疗': 19})

In [10]:
train_dataset = Dataset.from_list(train_data.data)
test_dataset = Dataset.from_list(test_data.data)
dev_dataset = Dataset.from_list(dev_data.data)

# 2.构造训练数据

In [11]:
def preprocess_function(tokenizer, examples):
    MAX_LENGTH = 512
    system_prompt = "你是一个中医药领域的命名体识别专家，你需要从给定句子中提取相关命名体，以json格式输出，例如：[['口苦', '临床表现'],['口干', '临床表现']]，其中每个元素分别表示命名体的起始位置、结束位置、文本内容和标签类型。没有实体则返回空列表[]"
    messages = [
        {"role": "system", "content": system_prompt},
        {"role": "user", "content": examples['input']},
    ]
    messages = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True,) # 输出会停在最后一个user，然后添加assistant的开始标记等待模型生成, 不会编码assistant的内容
    # print(messages)
    # print(examples)
    response = tokenizer(examples['output'], add_special_tokens=False)
    inputs = tokenizer(messages, truncation=True, add_special_tokens=False)

    input  = inputs['input_ids'] + response['input_ids'] + [tokenizer.pad_token_id]  # 将输入和输出拼接在一起，并添加结束标记

    attention_mask = inputs['attention_mask'] + response['attention_mask'] + [1]  # 拼接注意力掩码

    outputs = [-100] * len(inputs['input_ids']) + response['input_ids'] + [tokenizer.eos_token_id]  # 输出标签的token id，并添加结束标记

    if len(input) > MAX_LENGTH:
        input = input[:MAX_LENGTH]
        attention_mask = attention_mask[:MAX_LENGTH]
        outputs = outputs[:MAX_LENGTH]

    return {'input_ids': input, 'attention_mask': attention_mask, 'labels': outputs}

p_preprocess = partial(preprocess_function, tokenizer)

messages:
```<|im_start|>system
你是一个中医药领域的命名体识别专家，你需要从给定句子中提取相关命名体，以json格式输出，例如：[['口苦', '临床表现'],['口干', '临床表现']]，其中每个元素分别表示命名体的起始位置、结束位置、文本内容和标签类型。<|im_end|>
<|im_start|>user
现头昏口苦<|im_end|>
<|im_start|>assistant
```

In [14]:
print(train_dataset[0])
res = preprocess_function(tokenizer, train_dataset[0])
res

{'input': '现头昏口苦', 'output': '[["口苦", "临床表现"]]'}


{'input_ids': [151644,
  8948,
  198,
  56568,
  101909,
  107498,
  104799,
  109734,
  31914,
  102450,
  101057,
  3837,
  112735,
  45181,
  89012,
  22382,
  109949,
  15946,
  107439,
  78556,
  109734,
  31914,
  3837,
  23031,
  2236,
  68805,
  66017,
  3837,
  77557,
  5122,
  56330,
  39426,
  99746,
  516,
  364,
  104595,
  101107,
  84000,
  39426,
  99251,
  516,
  364,
  104595,
  101107,
  30840,
  3837,
  90919,
  103991,
  102268,
  101127,
  51463,
  109734,
  31914,
  9370,
  71618,
  26606,
  81812,
  5373,
  80565,
  81812,
  5373,
  108704,
  43815,
  33108,
  105151,
  31905,
  1773,
  80443,
  101565,
  46448,
  31526,
  34794,
  44177,
  1294,
  151645,
  198,
  151644,
  872,
  198,
  46451,
  64355,
  101616,
  39426,
  99746,
  151645,
  198,
  151644,
  77091,
  198,
  58,
  1183,
  39426,
  99746,
  497,
  330,
  104595,
  101107,
  66294,
  151643],
 'attention_mask': [1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  

In [19]:
''.join(map(str, res['attention_mask']))

'111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111'

In [13]:
tokenizer.decode(preprocess_function(tokenizer, train_dataset[0])['input_ids'])

'<|im_start|>system\n你是一个中医药领域的命名体识别专家，你需要从给定句子中提取相关命名体，以json格式输出，例如：[[\'口苦\', \'临床表现\'],[\'口干\', \'临床表现\']]，其中每个元素分别表示命名体的起始位置、结束位置、文本内容和标签类型。没有实体则返回空列表[]<|im_end|>\n<|im_start|>user\n现头昏口苦<|im_end|>\n<|im_start|>assistant\n[["口苦", "临床表现"]]<|endoftext|>'

In [12]:
train_datasets = train_dataset.map(p_preprocess, remove_columns=train_dataset.column_names, num_proc=4)
# test_datasets = test_dataset.map(p_preprocess, remove_columns=test_dataset.column_names, num_proc=4)
dev_datasets = dev_dataset.map(p_preprocess, remove_columns=dev_dataset.column_names, num_proc=4)

Map (num_proc=4):   0%|          | 0/5259 [00:00<?, ? examples/s]

Map (num_proc=4):   0%|          | 0/657 [00:00<?, ? examples/s]

#  3.训练

In [13]:
fine_tuning_config = LoraConfig(task_type=TaskType.CAUSAL_LM, 
                                target_modules=["q_proj", "k_proj", "v_proj", "o_proj", "gate_proj", "up_proj", "down_proj"],
                                r=8, 
                                lora_alpha=16, 
                                inference_mode=False, 
                                lora_dropout=0.1)

peft_model = get_peft_model(model, fine_tuning_config)
peft_model.print_trainable_parameters()

trainable params: 9,232,384 || all params: 1,552,946,688 || trainable%: 0.5945


In [14]:
peft_model

PeftModelForCausalLM(
  (base_model): LoraModel(
    (model): Qwen2ForCausalLM(
      (model): Qwen2Model(
        (embed_tokens): Embedding(151936, 1536)
        (layers): ModuleList(
          (0-27): 28 x Qwen2DecoderLayer(
            (self_attn): Qwen2Attention(
              (q_proj): lora.Linear(
                (base_layer): Linear(in_features=1536, out_features=1536, bias=True)
                (lora_dropout): ModuleDict(
                  (default): Dropout(p=0.1, inplace=False)
                )
                (lora_A): ModuleDict(
                  (default): Linear(in_features=1536, out_features=8, bias=False)
                )
                (lora_B): ModuleDict(
                  (default): Linear(in_features=8, out_features=1536, bias=False)
                )
                (lora_embedding_A): ParameterDict()
                (lora_embedding_B): ParameterDict()
                (lora_magnitude_vector): ModuleDict()
              )
              (k_proj): lora.Linear(
  

In [15]:
args = TrainingArguments(
    output_dir="G:/model_weights/models/checkpoint/qwen2.5-1.5-tcm-ner2",
    per_device_train_batch_size=4,
    per_device_eval_batch_size=4,
    gradient_accumulation_steps=4, # 积累多少batch后更新一次权重，主要用于解决显存不足的问题
    eval_strategy="epoch",
    logging_steps=10,
    num_train_epochs=2,
    save_steps=100,
    learning_rate=1e-4,
    save_on_each_node=True,
    gradient_checkpointing=True,
    report_to="none",
)

In [ ]:
n1lKeqMVw5Ur7zirkQJlq

In [16]:
swanlab_callback = SwanLabCallback(
    project="Qwen2.5-NER-fintune",
    experiment_name="Qwen2-1.5B-Instruct",
    description="使用通义千问Qwen2-1.5B-Instruct模型在中医药NER数据集上微调，实现关键实体识别任务。",
    config={
        "model": model_id,
        "model_dir": model_path,
        "dataset": "qgyd2021/chinese_ner_sft",
    },
)

In [17]:
trainer = Trainer(
    model=peft_model,
    args=args,
    train_dataset=train_datasets,
    eval_dataset=dev_datasets,
    data_collator=DataCollatorForSeq2Seq(tokenizer=tokenizer, padding=True),
    callbacks=[swanlab_callback],
)


In [18]:
trainer.train()

Output()

Output()

swanlab: Tracking run with swanlab version 0.7.8

swanlab: Run data will be saved locally in 
d:\doc\my_project\lianxi\home_work\stage_3\swanlog\run-20260222_102538-yjrk3ssohwo4mkcse3ajf

swanlab: 👋 Hi yuerg1230,welcome to swanlab!

swanlab: Syncing run Qwen2-1.5B-Instruct to the cloud

swanlab: 🏠 View project at https://swanlab.cn/@yuerg1230/Qwen2.5-NER-fintune

swanlab: 🚀 View run at https://swanlab.cn/@yuerg1230/Qwen2.5-NER-fintune/runs/yjrk3ssohwo4mkcse3ajf

Epoch,Training Loss,Validation Loss
1,0.126604,0.122962
2,0.095028,0.110986


TrainOutput(global_step=658, training_loss=0.1349273352093972, metrics={'train_runtime': 801.0203, 'train_samples_per_second': 13.131, 'train_steps_per_second': 0.821, 'total_flos': 1.467532752873984e+16, 'train_loss': 0.1349273352093972, 'epoch': 2.0})

样本数量是5259，训练的时候进度提示总共是658，原因？\
`per_device_train_batch_size` = 4 每次训练4个样本 \
`gradient_accumulation_steps` = 4 每4个batch更新一次权重 \
`num_train_epochs`=2              总共训练两轮 \
训练器显示的步数是权重更新的次数\
$5259 \div 4 \div 4 \times 2 \approx 658$

In [21]:
swanlab.finish()

swanlab: 🏠 View project at https://swanlab.cn/@yuerg1230/Qwen2.5-NER-fintune

swanlab: 🚀 View run at https://swanlab.cn/@yuerg1230/Qwen2.5-NER-fintune/runs/ily2vcbjlf3ixxfg333l2

# 4.加载检查点Lora

In [21]:
from peft import  PeftModel
base_model = AutoModelForCausalLM.from_pretrained(model_path).cuda()
tokenizer = AutoTokenizer.from_pretrained(model_path)
peft_model = PeftModel.from_pretrained(base_model, "G:/model_weights/models/checkpoint/qwen2.5-1.5-tcm-ner/checkpoint-658").cuda()

Loading weights:   0%|          | 0/338 [00:00<?, ?it/s]

In [22]:
from pathlib import Path
test_data_path = Path(data_path) / 'medical.test'

test_data = TCMNERDataset(test_data_path)

In [23]:
def f1_score(set_a, set_b):
    """ 严格的F1 分数计算，只有当索引位置、实体、标签预测完全正确时才算TP，否则算FP和FN """
    TP = len(set_a & set_b)
    FP = len(set_b - set_a)
    FN = len(set_a - set_b)
    
    precision = TP / (TP + FP) if (TP + FP) > 0 else 0
    recall = TP / (TP + FN) if (TP + FN) > 0 else 0
    f1 = 2 * precision * recall / (precision + recall) if (precision + recall) > 0 else 0
    return f1


## 批量样本推断

In [32]:
set_a_strict = set()
set_b_strict = set()
set_a_relax = set()
set_b_relax = set()


tag_dict = {}
for entity in test_data.label_set:
    tag_dict[entity] = [set(), set()]  # [label, label_predicted]

def predict_collect(predictions, references):
    """ 预测结果收集函数，返回严格匹配和宽松匹配的实体集合 """
    set_a_strict = set()  # 严格匹配：索引位置、实体内容、标签类型都必须完全匹配
    set_b_strict = set()  # 模型预测的严格匹配结果
    
    for item in references:
        item = json.loads(item)
        for label in item:
            print(label)
            tag_dict[label[1]][0].add(tuple(label))  # 仅比较实体内容和标签类型，忽略位置
            set_a_strict.add(tuple(label))

    for output in predictions:
        try:
            output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格

            for item in output:
                if item[1] in tag_dict:
                    tag_dict[item[1]][1].add(tuple(item))
                set_b_strict.add(tuple(item))
        except:
            import traceback
            print(references)
            print('--1')
            print(output)
            print('--2')
            traceback.print_exc()
    
    return set_a_strict, set_b_strict,

In [25]:
tag_dict

{'中药': [set(), set()],
 '西医诊断': [set(), set()],
 '中医证候': [set(), set()],
 '方剂': [set(), set()],
 '中医诊断': [set(), set()],
 '临床表现': [set(), set()],
 '中医治则': [set(), set()],
 '中医治疗': [set(), set()],
 '西医治疗': [set(), set()],
 '其他治疗': [set(), set()]}

In [33]:
import tqdm

from torch.utils.data import DataLoader

test_dataset = DataLoader(test_data.data, batch_size=32, shuffle=False)

system_prompt = '你是一个中医药领域的命名体识别专家，你需要从给定句子中提取相关命名体，以json格式输出，例如：[["口苦", "临床表现"],["口干", "临床表现"]]，其中每个元素分别表示实体和标签类型。如果检测不到实体则返回空列表[]'

pbar = tqdm.tqdm(total=len(test_dataset), desc="Evaluating on test set")

for batch_sample in test_dataset:
    inputs = []
    for i in batch_sample['input']:
        messages = [
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": i},
        ]
        input = tokenizer.apply_chat_template(messages, padding_side='left', tokenize=False, add_generation_prompt=True) # 输出会停在最后一个user，然后添加assistant的开始标记等待模型生成, 不会编码assistant的内容
        inputs.append(input)
    inputs = tokenizer(inputs, padding_side='left', padding=True, truncation=True, return_tensors="pt").to(peft_model.device)

    with torch.no_grad():
        outputs = peft_model.generate(inputs.input_ids, max_new_tokens=512)
    
    output = outputs[:, inputs.input_ids.shape[1]:]  # 获取生成的部分
    decoded_outputs = tokenizer.batch_decode(output, skip_special_tokens=True)

    set_a_strict, set_b_strict = predict_collect(decoded_outputs, batch_sample['output'])

Evaluating on test set:   0%|          | 0/21 [08:44<?, ?it/s]


['黄疸', '中医诊断']
['法莫替丁', '西医治疗']
['胃食管反流病', '西医诊断']
['疏肝行气', '中医治则']
['调神解郁', '中医治则']
['推拿', '中医治疗']
['腹泻', '临床表现']
['腹痛', '临床表现']
['功能性便秘', '西医诊断']
['头晕', '临床表现']
['腹胀胁满', '临床表现']
['大便溏薄', '临床表现']
['舌苔薄黄腻', '临床表现']
['脉细微弦', '临床表现']
['七方胃痛胶囊', '中医治疗']
['肝郁脾虚', '中医证候']
['腹泻', '临床表现']
['枳术丸加味', '方剂']
['功能性便秘', '西医诊断']
['大便', '临床表现']
['舌苔黄腻', '临床表现']
['瘀毒根深', '中医证候']
['疏香灸', '中医治疗']
['便秘型肠易激综合征', '西医诊断']
['当归', '中药']
['痛泻要方加味', '方剂']
['心理疗法', '其他治疗']
['金银花', '中药']
['连翘', '中药']
['牛蒡子', '中药']
['荆芥', '中药']
['防风', '中药']
['藿香', '中药']
['佩兰', '中药']
['碧玉散', '中药']
['桔梗', '中药']
['枳壳', '中药']
['身面发黄', '临床表现']
['目黄如金', '临床表现']
['脘痞纳差', '临床表现']
['厌食泛恶', '临床表现']
['口苦', '临床表现']
['尿如柏汁', '临床表现']
['精神萎顿', '临床表现']
['胸胁疼痛', '临床表现']
['苔黄厚', '临床表现']
['腹泻型肠易激综合征', '西医诊断']
['头晕', '临床表现']
['穴位埋线', '中医治疗']
['肝郁气滞', '中医证候']
['血活素', '中医治疗']
['非酒精性脂肪肝', '西医诊断']
['急黄', '中医诊断']
['毒陷心包', '中医证候']
['香砂枳术颗粒', '中医治疗']
['脾胃虚弱', '中医证候']
['针刺', '中医治疗']
['功能性消化不良', '西医诊断']
['针刺', '中医治疗']
['针刺', '中医治疗']
['功能性消化不良', '西医诊断']
['子午流注

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 98 (char 97)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppD

['和胃冲剂', '方剂']
['肝胃不和', '中医证候']
['功能性消化不良', '西医诊断']
['腹胀', '临床表现']
['尿少', '临床表现']
['恶心呕吐', '临床表现']
['苔黄', '临床表现']
['肝内小结石', '西医诊断']
['肝硬化腹水', '西医诊断']
['积聚', '中医诊断']
['臌胀', '中医诊断']
['半夏厚朴汤加味', '方剂']
['功能性消化不良', '西医诊断']
['超声离子导入', '其他治疗']
['腹痛', '临床表现']
['腹胀', '临床表现']
['脾胃虚弱', '中医证候']
['功能性消化不良', '西医诊断']
['香砂六君子汤', '方剂']
['贴敷', '中医治疗']
['香砂六君子汤', '方剂']
['针刺', '中医治疗']
['针刺', '中医治疗']
['慢性功能性便秘', '西医诊断']
['儿童功能性消化不良', '西医诊断']
['滋阴润肠口服液', '中医治疗']
['老年功能性便', '西医诊断']
['脾胃气虚', '中医证候']
['功能性消化不良', '西医诊断']
['梅花灸', '中医治疗']
['穴位贴敷', '中医治疗']
['功能性消化不良', '西医诊断']
['痰瘀互结', '中医证候']
['非酒精性脂肪肝', '西医诊断']
['退黄', '中医治则']
['浮肿', '临床表现']
['中风后遗症功能性便秘', '西医诊断']
['清肝和胃饮', '方剂']
['马来酸曲美布汀片', '西医治疗']
['肝胃郁热', '中医证候']
['功能性消化不良', '西医诊断']
['健胃消胀颗粒', '中医治疗']
['功能性消化不良', '西医诊断']
['丝瓜络', '中药']
['增液承气汤加减', '方剂']
['喘脱', '中医诊断']
['水鼓', '中医诊断']
['黄疸', '中医诊断']
['脾胃湿热', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['肠易激综合征', '西医诊断']
['非酒精性脂肪肝', '西医诊断']
['水臌', '中医诊断']
['益肠汤', '方剂']
['柴胡', '中药']
['党参', '中药']
['白术', '中药']
['陈皮', '中药']
['茯苓', '中

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 28, in predict_collect
    if item[1] in tag_dict:
       ~~~~^^^
IndexError: list index out of range
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    

['小陷胸汤', '方剂']
['胃食管反流病', '西医诊断']
['温中健脾', '中医治则']
['脾胃虚寒', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['六味能消胶囊', '中医治疗']
['功能性消化不良', '西医诊断']
['肝硬化腹水', '西医诊断']
['功能性消化不良', '西医诊断']
['脾虚气滞', '中医证候']
['健脾理气方', '方剂']
['盐酸依托必利', '西医治疗']
['甘草', '中药']
['黄疸', '中医诊断']
['慢性萎缩性胃炎', '西医诊断']
['慢性萎缩性胃炎', '西医诊断']
['瑞巴派特', '西医治疗']
['贝洛纳', '西医治疗']
['小陷胸汤', '方剂']
['腹水', '临床表现']
['肠易激综合征', '西医诊断']
['腹部不适', '临床表现']
['便秘', '临床表现']
['腹痛', '临床表现']
['慢性萎缩性胃炎', '西医诊断']
['寒热错杂', '中医证候']
['急性重症肝炎', '西医诊断']
['身黄', '临床表现']
['肝细胞性黄疸', '西医诊断']
['腹水消失', '临床表现']
['胃痛', '临床表现']
['痞满', '临床表现']
['纳呆', '临床表现']
['脉数大无力', '临床表现']
['黄疸', '中医诊断']
['腹胀', '临床表现']
['黄疸', '中医诊断']
['柴胡', '中药']
['血瘀热毒', '中医证候']
['逍遥散', '方剂']
['功能性消化不良', '西医诊断']
['胃宁胶囊', '中医治疗']
['胃康宁', '中医治疗']
['寒热错杂', '中医证候']
['六味安消胶囊', '中医治疗']
['疏肝理气', '中医治则']
['功能性消化不良', '西医诊断']
['腹泻型肠易激综合征', '西医诊断']
['健胃消胀颗粒', '中医治疗']
['功能性消化不良', '西医诊断']
['肝胃不和', '中医证候']
['健胃消胀颗粒', '中医治疗']
['功能性消化不良', '西医诊断']
['胃食管反流性咽喉炎', '西医诊断']
['[["小陷胸汤", "方剂"]]', '[["胃食管反流病", "西医诊断"]]', '[["温中健脾", "中医治则"

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ip

['脾肾阳虚', '中医证候']
['急黄', '中医诊断']
['鼓胀', '中医诊断']
['功能性消化不良', '西医诊断']
['温针灸', '中医治疗']
['恶寒', '临床表现']
['疏肝化瘀', '中医治则']
['面色黧黑', '临床表现']
['巩膜重度黄染', '临床表现']
['厌食', '临床表现']
['小便深黄', '临床表现']
['大便秘结', '临床表现']
['苔白腻', '临床表现']
['脉细', '临床表现']
['针灸', '中医治疗']
['老年慢性功能性便秘', '西医诊断']
['桃核承气汤加味', '方剂']
['抵当丸', '方剂']
['黄疸渐退', '临床表现']
['食欲渐增', '临床表现']
['中老年功能性便秘', '西医诊断']
['肝胃不和', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['慢性萎缩性胃炎', '西医诊断']
['寒热错杂', '中医证候']
['乌梅丸', '方剂']
['中药敷贴', '中医治疗']
['功能性便秘', '西医诊断']
['穴位埋线', '中医治疗']
['食欲不振', '临床表现']
['肝区疼痛', '临床表现']
['党参', '中药']
['炒白术', '中药']
['炒柴胡', '中药']
['炙黄芪', '中药']
['当归', '中药']
['酸枣仁', '中药']
['陈皮', '中药']
['肉桂', '中药']
['炒白芍', '中药']
['大枣', '中药']
['赤小豆', '中药']
['炙甘草', '中药']
['生姜', '中药']
['枳术丸', '方剂']
['中老年慢性功能性便秘', '西医诊断']
['耳穴贴压', '中医治疗']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['奥美拉唑', '西医治疗']
['多潘立酮', '西医治疗']
['滋阴养血润肠汤', '方剂']
['津亏血少', '中医证候']
['老年功能性便秘', '西医诊断']
['当归', '中药']
['慢性萎缩性胃炎', '西医诊断']
['四逆散加味', '方剂']
['慢性功能性便秘', '西医诊断']
['肠道气滞', '中医证候']
['四逆散', '方剂']
['气滞血瘀', '中医

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ip

['慢性萎缩性胃炎', '西医诊断']
['急性黄疸型肝炎', '西医诊断']
['寒湿郁遏发黄', '中医证候']
['党参', '中药']
['慢性萎缩性胃炎伴异型增生', '西医诊断']
['健脾消积颗粒', '中医治疗']
['脾胃湿热', '中医证候']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['柴胡疏肝散', '方剂']
['枳术丸加减', '方剂']
['肝胃不和', '中医证候']
['功能性消化不良', '西医诊断']
['行气活血', '中医治则']
['健脾利水', '中医治则']
['功能性消化不良', '西医诊断']
['枳术宽中胶囊', '中医治疗']
['非酒精性脂肪肝', '西医诊断']
['多烯磷脂酰胆碱', '西医治疗']
['功能性消化不良', '西医诊断']
['胃复方', '方剂']
['脾虚气滞', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['功能性消化不良', '西医诊断']
['肝郁脾虚', '中医证候']
['加味逍遥散', '方剂']
['肝郁脾虚', '中医证候']
['苍术', '中药']
['脾虚气滞', '中医证候']
['功能性消化不良', '西医诊断']
['小陷胸汤', '方剂']
['胃食管反流病', '西医诊断']
['非酒精性脂肪肝', '西医诊断']
['安络化纤丸', '中医治疗']
['面色萎黄不鲜', '临床表现']
['目珠黄染', '临床表现']
['小便黄赤', '临床表现']
['大便淡黄有时灰白', '临床表现']
['腹部膨胀', '临床表现']
['精神不振', '临床表现']
['喂乳后欲呕', '临床表现']
['苔薄腻', '临床表现']
['黄疸', '临床表现']
['足肿', '临床表现']
['小便短少', '临床表现']
['便秘型肠易激综合征', '西医诊断']
['肝郁气滞', '中医证候']
['逍遥散', '方剂']
['老十针', '中医治疗']
['肝气郁结', '中医证候']
['餐后不适综合征', '西医诊断']
['赤芍', '中药']
['功能性消化不良', '西医诊断']
['寒热错杂', '中医证候']
['腹水', '临床表现']
['腹胀', '临床表现']
['饮食增',

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 75 (char 74)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppD

['脾胃湿热', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['乏力', '临床表现']
['胃康乐胶囊Ⅱ号', '中医治疗']
['香附', '中药']
['砂仁', '中药']
['山药', '中药']
['党参', '中药']
['佛手', '中药']
['肠易激综合征', '西医诊断']
['胸腹、肢体肌肉已见明显增多', '临床表现']
['慢性萎缩性胃炎', '西医诊断']
['夏连薏英汤', '方剂']
['慢性萎缩性胃炎癌前病变', '西医诊断']
['湿热中阻', '中医证候']
['肝功能正常', '临床表现']
['奥美拉唑', '西医治疗']
['胃食管反流性咽喉炎', '西医诊断']
['疏肝健脾', '中医治则']
['中医情志调理', '中医治疗']
['焦虑', '西医诊断']
['非酒精性脂肪肝', '西医诊断']
['慢性萎缩性胃炎', '西医诊断']
['儿童功能性便秘', '西医诊断']
['推拿', '中医治疗']
['生物反馈治疗', '其他治疗']
['生白术', '中药']
['右上腹疼痛', '临床表现']
['呕吐不食', '临床表现']
['口苦', '临床表现']
['腹胀', '临床表现']
['低热', '临床表现']
['大便秘结', '临床表现']
['舌苔黄厚腻', '临床表现']
['脉弦滑', '临床表现']
['功能性消化不良', '西医诊断']
['麻黄', '中药']
['桑皮', '中药']
['郁金', '中药']
['慢性萎缩性胃炎', '西医诊断']
['肝胃郁热', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['自拟疏肝通腑汤', '方剂']
['气滞', '中医证候']
['功能性便秘', '西医诊断']
['建中振萎汤', '方剂']
['瘀阻胃络', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['功能性消化不良', '西医诊断']
['慢性胃炎', '西医诊断']
['胃下垂', '西医诊断']
['肠道津亏', '中医证候']
['苔白厚', '临床表现']
['慢性萎缩性胃炎', '西医诊断']
['气滞血瘀', '中医证候']
['莫沙必利', '西医治疗']
['选时穴位按摩', '中医治疗']
['便秘型肠易激综合征', '

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ip

['自拟化痰行瘀方', '方剂']
['非酒精性脂肪肝', '西医诊断']
['复方丁香开胃贴', '中医治疗']
['心功能不全伴功能性消化不良', '西医诊断']
['腹泻型肠易激综合征', '西医诊断']
['肠康方', '中医治疗']
['腹泻型肠易激综合征', '西医诊断']
['黄疸', '临床表现']
['腹胀', '临床表现']
['半夏泻心汤', '方剂']
['胃食管反流性咳嗽', '西医诊断']
['穴位埋线', '中医治疗']
['脾胃虚寒', '中医证候']
['功能性消化不良', '西医诊断']
['肝郁脾虚', '中医证候']
['柴芍六君子汤', '方剂']
['半夏泻心汤', '方剂']
['腹泻型肠易激综合征', '西医诊断']
['贝洛纳', '西医治疗']
['泰胃美', '西医治疗']
['胃食管反流性疾病', '西医诊断']
['肝胆气郁', '中医证候']
['香附', '中药']
['黄疸', '中医诊断']
['寒湿型', '中医证候']
['腹水已消', '临床表现']
['木香流气饮', '方剂']
['茵陈', '中药']
['炙鳖甲', '中药']
['猪苓', '中药']
['功能性消化不良', '西医诊断']
['腹部火龙疗法', '中医治疗']
['心理疏导', '其他治疗']
['脾胃气虚', '中医证候']
['功能性消化不良', '西医诊断']
['乌梅丸', '方剂']
['中药敷贴', '中医治疗']
['寒热错杂', '中医证候']
['功能性消化不良', '西医诊断']
['面色咣白', '临床表现']
['浮肿', '临床表现']
['形寒畏冷', '临床表现']
['倦怠无力', '临床表现']
['饥不欲食', '临床表现']
['胆怯', '临床表现']
['易惊', '临床表现']
['肝区偶觉刺痛', '临床表现']
['小便或清或黄', '临床表现']
['大便溏薄', '临床表现']
['电针', '中医治疗']
['功能性便秘', '西医诊断']
['脾胃虚弱', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['胰酶肠溶胶囊', '西医治疗']
['多潘立酮片', '西医治疗']
['食后腹胀', '临床表现']
['早饱', '临床表现']
['胃脘疼痛',

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 7 (char 6)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppDat

['脾虚气滞', '中医证候']
['功能性消化不良', '西医诊断']
['开胃进食汤超微配方颗粒', '中医治疗']
['痞满证', '中医诊断']
['慢性浅表性胃炎', '西医诊断']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['脾胃虚寒', '中医证候']
['韦氏整脊手法', '中医治疗']
['解毒扶正治萎方', '方剂']
['党参', '中药']
['白术', '中药']
['黄芪', '中药']
['蒲公英', '中药']
['重楼', '中药']
['丹参', '中药']
['头昏', '临床表现']
['腹胀', '临床表现']
['手心发热', '临床表现']
['呃逆', '中医诊断']
['食后腹胀', '临床表现']
['早饱', '临床表现']
['胃脘疼痛', '临床表现']
['烧心感', '临床表现']
['嗳气', '临床表现']
['恶心呕吐', '临床表现']
['胸胁胀痛', '临床表现']
['功能性消化不良', '西医诊断']
['脾虚气滞', '中医证候']
['功能性消化不良', '西医诊断']
['针刺', '中医治疗']
['慢性萎缩性胃炎', '西医诊断']
['腹泻', '临床表现']
['腹痛', '临床表现']
['腹胀', '临床表现']
['肛门下坠感', '临床表现']
['枳术宽中胶囊', '中医治疗']
['功能性消化不良', '西医诊断']
['胃食管反流性咳嗽', '西医诊断']
['咳嗽', '临床表现']
['嗳气', '临床表现']
['瑞舒伐他汀', '西医治疗']
['非酒精性脂肪肝', '西医诊断']
['功能性消化不良', '西医诊断']
['津亏肠燥气滞', '中医证候']
['便秘', '西医诊断']
['助阳通便汤', '方剂']
['便秘', '西医诊断']
['肝病治疗仪', '西医治疗']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['难寐', '临床表现']
['复方畅胃宁', '中医治疗']
['枳壳', '中药']
['佛手', '中药']
['柴胡', '中药']
['砂仁', '中药']
['生山楂', '中药']
['生麦芽', '中药']
['功能性消化不良', '西医

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 7 (char 6)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppDat

['炙甘草', '中药']
['急性黄疸型病毒性乙型肝炎', '西医诊断']
['非酒精性脂肪肝', '西医诊断']
['生活方式干预', '其他治疗']
['推拿', '中医治疗']
['面色晦暗', '临床表现']
['舌质暗红', '临床表现']
['舌紫有瘀斑', '临床表现']
['气滞血瘀', '中医证候']
['功能性消化不良', '西医诊断']
['上腹痛', '临床表现']
['早饱', '临床表现']
['嗳气', '临床表现']
['烧心', '临床表现']
['慢性萎缩性胃炎', '西医诊断']
['脾胃虚寒', '中医证候']
['腹水', '临床表现']
['当归', '中药']
['匹维溴铵', '西医治疗']
['匹维溴铵', '西医治疗']
['功能性便秘', '西医诊断']
['形体虚弱', '临床表现']
['黄疸', '中医诊断']
['头痛', '临床表现']
['右肋隐痛', '临床表现']
['体倦', '临床表现']
['口苦', '临床表现']
['口淡', '临床表现']
['尿黄', '临床表现']
['苔白厚', '临床表现']
['化浊颗粒', '中医治疗']
['茵陈', '中药']
['茵陈', '中药']
['栀子', '中药']
['草蔻', '中药']
['薏米', '中药']
['藿香', '中药']
['川厚朴', '中药']
['半夏', '中药']
['滑石', '中药']
['车前子', '中药']
['甘草', '中药']
['针刺', '中医治疗']
['脾气亏', '中医诊断']
['心理干预', '其他治疗']
['功能性消化不良', '西医诊断']
['扶正理气汤', '方剂']
['党参', '中药']
['甘草', '中药']
['陈皮', '中药']
['台乌', '中药']
['当归', '中药']
['桃仁', '中药']
['炒谷麦芽', '中药']
['黄疸渐退', '临床表现']
['寒湿夹瘀', '中医证候']
['阴黄', '中医诊断']
['隔药灸', '中医治疗']
['腹泻型肠易激综合征', '西医诊断']
['糖尿病性功能性消化不良', '西医诊断']
['荆芥', '中药']
['柴胡', '中药']
['肝胃不和', '中医证候']
['功能性消化

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 9 (char 8)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppDat

['儿童功能性便秘', '西医诊断']
['目黄', '临床表现']
['尿黄如浓茶', '临床表现']
['精神', '临床表现']
['食欲可', '临床表现']
['口不苦', '临床表现']
['苔白滑', '临床表现']
['脉细', '临床表现']
['辗转不安', '临床表现']
['大承气汤', '方剂']
['耳穴贴压', '中医治疗']
['枳术宽中胶囊', '中医治疗']
['脂肝合剂', '方剂']
['肝郁脾虚', '中医证候']
['枳术宽中胶囊', '中医治疗']
['功能性消化不良', '西医诊断']
['多烯磷脂酰胆碱', '西医治疗']
['高压氧', '西医治疗']
['围绝经期功能性消化不良', '西医诊断']
['功能性消化不良伴厌食症', '西医诊断']
['推拿', '中医治疗']
['益气开秘方', '方剂']
['四神丸', '方剂']
['脾肾阳虚', '中医证候']
['腹泻型肠易激综合征', '西医诊断']
['苔薄白', '临床表现']
['食欲不振', '临床表现']
['恶心', '临床表现']
['湿热疫毒', '中医证候']
['枯草杆菌二联活菌肠溶胶囊', '西医治疗']
['腹泻型肠易激综合征', '西医诊断']
['烧心', '临床表现']
['反酸', '临床表现']
['反食', '临床表现']
['胸骨后灼痛', '临床表现']
['茵陈', '中药']
['功能性消化不良', '西医诊断']
['萎缩性胃炎', '西医诊断']
['脾胃虚弱', '中医证候']
['柴胡疏肝散', '方剂']
['阴黄', '中医诊断']
['针刺疗法', '中医治疗']
['功能性便秘', '西医诊断']
['茯苓', '中药']
['腹泻型肠易激综合征', '西医诊断']
['肉桂', '中药']
['功能性消化不良', '西医诊断']
['肝郁脾虚', '中医证候']
['四逆散加味', '方剂']
['性急', '临床表现']
['心烦', '临床表现']
['少腹逐瘀汤', '方剂']
['大黄', '中药']
['白芍', '中药']
['三七粉', '中药']
['健脾疏肝', '中医治则']
['腹水', '临床表现']
['肝郁脾虚', '中医证候']
['葶苈子', '中药']
['腹

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ip

['四联疗法', '其他治疗']
['小便增多', '临床表现']
['多烯磷脂酰胆碱', '西医治疗']
['非酒精性脂肪肝', '西医诊断']
['雷贝拉唑胶囊', '西医治疗']
['功能性消化不良', '西医诊断']
['紫绀', '临床表现']
['白芍', '中药']
['建中振萎汤', '方剂']
['慢性萎缩性胃炎', '西医诊断']
['脾胃虚弱', '中医证候']
['功能性消化不良', '西医诊断']
['恶寒', '临床表现']
['黄疸', '中医诊断']
['头痛', '临床表现']
['口苦', '临床表现']
['尿黄', '临床表现']
['苔白厚', '临床表现']
['参柴胃苏胶囊', '中医治疗']
['功能性消化不良', '西医诊断']
['四逆散加味', '方剂']
['肝郁脾虚', '中医证候']
['功能性消化不良', '西医诊断']
['恶心欲吐', '临床表现']
['脘腹胀痞', '临床表现']
['舒肝和胃饮', '方剂']
['功能性消化不良', '西医诊断']
['路优泰', '西医治疗']
['便秘', '中医诊断']
['奥美拉唑和莫沙必利', '西医治疗']
['胃食管反流病', '西医诊断']
['苓桂术甘汤加味', '方剂']
['茯苓', '中药']
['桂枝', '中药']
['炙甘草', '中药']
['附子', '中药']
['竹茹', '中药']
['木通', '中药']
['茅根', '中药']
['姜皮', '中药']
['法半夏', '中药']
['风湿性心脏病', '西医诊断']
['疏肝通络', '中医治则']
['健脾利水', '中医治则']
['扶正', '中医治则']
['当归芍药散', '方剂']
['阴黄', '中医诊断']
['脾虚兼有湿热', '中医证候']
['健脾兼清利湿热', '中医治则']
['痰热内壅', '中医证候']
['清肺化痰', '中医治则']
['功能性消化不良', '西医诊断']
['温通针法', '中医治疗']
['湿热蕴结', '中医证候']
['胎黄', '中医诊断']
['茵陈蒿汤加味', '方剂']
['茵陈', '中药']
['栀子', '中药']
['制大黄', '中药']
['赤芍', '中药']
['茯苓', '中药']


Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)


['功能性消化不良', '西医诊断']
['抑郁', '临床表现']
['焦虑', '临床表现']
['调肝理脾通腑', '中医治则']
['便秘型肠易激综合征', '西医诊断']
['腰痛', '临床表现']
['白术芪蓉汤', '方剂']
['脾肾两虚', '中医证候']
['老年功能性便秘', '西医诊断']
['慢性萎缩性胃炎', '西医诊断']
['自拟胃宁煎剂', '方剂']
['黄芪', '中药']
['党参', '中药']
['茯苓', '中药']
['香附', '中药']
['佛手', '中药']
['柴胡桂枝汤加味', '方剂']
['柴胡', '中药']
['川芎', '中药']
['半夏', '中药']
['黄芩', '中药']
['葛根', '中药']
['甘草', '中药']
['干姜', '中药']
['黄连', '中药']
['温阳利水', '中医治则']
['腹水', '临床表现']
['脾胃虚寒', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['埋线疗法', '中医治疗']
['川楝', '中药']
['非酒精性脂肪肝', '西医诊断']
['厄贝沙坦', '西医治疗']
['二甲双胍', '西医治疗']
['疏肝健脾和胃', '中医治则']
['归芍六君子汤', '方剂']
['当归', '中药']
['白芍', '中药']
['陈皮', '中药']
['半夏', '中药']
['党参', '中药']
['白术', '中药']
['茯苓', '中药']
['电针', '中医治疗']
['自主神经功能', '西医诊断']
['心理状态', '西医诊断']
['穴位埋线', '中医治疗']
['功能性便秘', '西医诊断']
['盐酸依托必利', '西医治疗']
['功能性消化不良', '西医诊断']
['针刺', '中医治疗']
['脾胃虚弱', '中医证候']
['穴位埋线', '中医治疗']
['功能性便秘', '西医诊断']
['白术', '中药']
['麻痹性肠梗阻', '西医诊断']
['茵陈', '中药']
['胃和汤', '方剂']
['慢性萎缩性胃炎', '西医诊断']
['胃食管反流病', '西医诊断']
['中医穴位按摩', '中医治疗']
['慢性功能性便秘', '西医诊断']
['肝胆湿热',

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 13 (char 12)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppD

['肠易激综合征', '西医诊断']
['电针', '中医治疗']
['胃食管反流病', '西医诊断']
['穴位敷贴', '中医治疗']
['老年功能性便秘', '西医诊断']
['补中益气汤加减', '方剂']
['穴位敷贴', '中医治疗']
['脾胃气虚', '中医证候']
['功能性消化不良', '西医诊断']
['补肾健脾通便颗粒', '中医治疗']
['脾肾阳虚', '中医证候']
['慢性功能性便秘', '西医诊断']
['耳穴埋豆', '中医治疗']
['老年高血压患者功能性便秘', '西医诊断']
['穴位贴敷', '中医治疗']
['腹泻型肠易激综合征', '西医诊断']
['脾胃湿热', '中医证候']
['呕吐消失', '临床表现']
['腹胀', '临床表现']
['黄疸', '临床表现']
['泄泻消失', '临床表现']
['纳增', '临床表现']
['肝区疼痛', '临床表现']
['腹胀', '临床表现']
['功能性消化不良', '西医诊断']
['面色暗黄', '临床表现']
['形寒怕冷', '临床表现']
['精神疲惫', '临床表现']
['口淡', '临床表现']
['纳呆', '临床表现']
['动则汗出心慌', '临床表现']
['夜寐不实', '临床表现']
['大便溏薄', '临床表现']
['小便黄赤', '临床表现']
['冠心病', '西医诊断']
['体倦', '临床表现']
['大黄', '中药']
['腹软', '临床表现']
['痞满', '中医诊断']
['郁证', '中医诊断']
['胃脘痛', '中医诊断']
['肝胃不合', '中医证候']
['功能性消化不良', '西医诊断']
['舒肝和胃汤', '方剂']
['胃食管反流病', '西医诊断']
['肝郁脾虚', '中医证候']
['功能性消化不良', '西医诊断']
['化滞柔肝颗粒', '中医治疗']
['非酒精性脂肪肝', '西医诊断']
['功能性消化不良', '西医诊断']
['乏力', '临床表现']
['腹胀', '临床表现']
['功能性消化不良', '西医诊断']
['肝胃不和', '中医证候']
['便秘', '中医诊断']
['活血益胃汤', '方剂']
['苔薄黄', '临床表现']
['健脾舒肝汤', '方剂

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ip

['疳脂平片', '中医治疗']
['非酒精性脂肪肝', '西医诊断']
['四联疗法', '西医治疗']
['邪热内蕴', '中医证候']
['腑气不通', '中医证候']
['功能性便秘', '西医诊断']
['四神丸', '方剂']
['附子理中汤', '方剂']
['腹泻型肠易激综合征', '西医诊断']
['脾肾阳虚', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['身目发黄', '临床表现']
['心中烦闷', '临床表现']
['厌食泛恶', '临床表现']
['腹痛', '临床表现']
['腹泻', '临床表现']
['便若脓血', '临床表现']
['里急后重', '临床表现']
['肛门灼热', '临床表现']
['小便短少', '临床表现']
['苔黄腻', '临床表现']
['非酒精性脂肪肝', '西医诊断']
['复方甘草酸苷', '中医治疗']
['功能性消化不良', '西医诊断']
['上腹痛', '临床表现']
['上腹胀', '临床表现']
['早饱', '临床表现']
['嗳气', '临床表现']
['食欲不振', '临床表现']
['恶心', '临床表现']
['呕吐', '临床表现']
['针刺', '中医治疗']
['胃食管反流病', '西医诊断']
['香苏饮', '方剂']
['功能性消化不良', '西医诊断']
['舒肝和胃饮', '方剂']
['功能性消化不良', '西医诊断']
['肝胃不和', '中医证候']
['功能性消化不良', '西医诊断']
['食滞', '中医证候']
['功能性消化不良', '西医诊断']
['食滞', '中医证候']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['肝胃不和', '中医证候']
['蜥蜴胃康基本方', '方剂']
['慢性萎缩性胃炎', '西医诊断']
['气阴不足', '中医证候']
['毒瘀交阻', '中医证候']
['炒白术', '中药']
['加味逍遥散', '方剂']
['肝郁脾虚', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['胃食管反流病', '西医诊断']
['功能性消化不良', '西医诊断']
['肝脾气滞', '中医证候']
['急性黄疸型肝炎

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 129 (char 128)
Traceback (most recent call last):
  File "C:\Users\lkjx0\Ap

['电针治疗', '中医治疗']
['气虚', '中医证候']
['功能性便秘', '西医诊断']
['消瘦', '临床表现']
['精神不振', '临床表现']
['奥美拉唑和莫沙必利', '西医治疗']
['胃食管反流病', '西医诊断']
['黛力新', '西医治疗']
['功能性消化不良', '西医诊断']
['肝脾不和', '中医证候']
['解毒扶正治萎方', '方剂']
['慢性萎缩性胃炎', '西医诊断']
['腹部推拿', '中医治疗']
['黄疸', '中医诊断']
['发热', '中医诊断']
['湿热疫毒侵入血分', '中医证候']
['健脾疏肝降逆方', '方剂']
['功能性消化不良', '西医诊断']
['胃康乐胶囊Ⅱ号', '中医治疗']
['循经点穴推拿', '中医治疗']
['功能性消化不良', '西医诊断']
['针刺', '中医治疗']
['针刺', '中医治疗']
['便秘型肠易激综合征', '西医诊断']
['腹泻型功能性消化不良', '西医诊断']
['香砂六君子汤', '方剂']
['贴敷', '中医治疗']
['脾胃虚弱', '中医证候']
['功能性消化不良', '西医诊断']
['苔黄', '临床表现']
['压痛', '临床表现']
['脉象细数', '临床表现']
['耳穴压豆', '中医治疗']
['慢性萎缩性胃炎', '西医诊断']
['老年功能性便秘', '西医诊断']
['埃索美拉唑', '西医治疗']
['胃食管反流病', '西医诊断']
['苓桂术甘汤', '方剂']
['肝硬化', '西医诊断']
['慢性萎缩性胃炎', '西医诊断']
['脾胃虚弱', '中医证候']
['枸橼酸莫沙必利片', '西医治疗']
['腹痛', '临床表现']
['腹泻', '临床表现']
['黄连', '中药']
['功能性消化不良', '西医诊断']
['精神好转', '临床表现']
['腹胀痛消失', '临床表现']
['黄疸变淡', '临床表现']
['小便淡黄', '临床表现']
['大便黄', '临床表现']
['矢气多', '临床表现']
['舌正红', '临床表现']
['瘀点减少', '临床表现']
['苔黄厚腻', '临床表现']
['脉弦', '临床表现']
['腹胀', '临床表现']
['

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 75 (char 74)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppD

['胆道死蛔残体酿生热毒', '临床表现']
['水肿（阴水）', '中医诊断']
['脾肾两虚', '中医证候']
['面色黑', '临床表现']
['声音嘶哑', '临床表现']
['舌红', '临床表现']
['苔薄白', '临床表现']
['肝硬化腹水', '西医诊断']
['气滞水阻兼肝阴不足之证', '中医证候']
['脾胃虚弱', '中医证候']
['乙型肝炎后肝硬化失代偿期', '西医诊断']
['三仁汤', '方剂']
['脾胃湿热', '中医证候']
['功能性消化不良', '西医诊断']
['乏力', '临床表现']
['纳差', '临床表现']
['腹胀', '临床表现']
['眠差', '临床表现']
['二便尚调', '临床表现']
['面容虚胖', '临床表现']
['面色苍白', '临床表现']
['巩膜轻度黄染', '临床表现']
['苔白', '临床表现']
['脉弦细', '临床表现']
['老年功能性消化不良', '西医诊断']
['多潘立酮', '西医治疗']
['多潘立酮', '西医治疗']
['慢性萎缩性胃炎', '西医诊断']
['柴胡疏肝散', '方剂']
['穴位注射', '中医治疗']
['枳壳', '中药']
['功能性消化不良', '西医诊断']
['脾虚湿阻', '中医证候']
['脾阳虚', '中医证候']
['糖脂消', '中医治疗']
['非酒精性脂肪肝', '西医诊断']
['温针灸', '中医治疗']
['慢性萎缩性胃炎', '西医诊断']
['功能性消化不良', '西医诊断']
['枳术丸', '方剂']
['肠宁汤', '方剂']
['肠易激综合征', '西医诊断']
['异功散合二金汤加减', '方剂']
['上腹疼痛', '临床表现']
['上腹饱胀', '临床表现']
['纳差', '临床表现']
['嗳气', '临床表现']
['疏肝和胃', '中医治则']
['复方蜥蜴散不同微粒组合剂', '方剂']
['附片', '中药']
['干姜', '中药']
['上肉桂', '中药']
['茯苓', '中药']
['肝郁气滞', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['泽泻', '中药']
['黄疸', '中医诊断']
['破血', '中医治则']
['通经', '中医

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 7 (char 6)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppDat

['功能性消化不良', '西医诊断']
['二参三草汤', '方剂']
['慢性萎缩性胃炎', '西医诊断']
['脾虚气弱', '中医证候']
['口苦', '临床表现']
['黄疸', '中医诊断']
['慢性胃炎', '西医诊断']
['胃安胶囊', '中医治疗']
['功能性消化不良', '西医诊断']
['苔薄白', '临床表现']
['肝郁脾虚', '中医证候']
['功能性消化不良', '西医诊断']
['红细胞增多症', '西医诊断']
['肠宁汤', '方剂']
['新下气汤', '方剂']
['脾虚肝乘', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['板蓝根', '中药']
['连翘', '中药']
['甘草', '中药']
['慢性胃炎', '西医诊断']
['气血瘀滞', '中医证候']
['湿热', '中医证候']
['黄疸', '中医诊断']
['湿热', '中医证候']
['冠心病', '西医诊断']
['慢性上腹疼痛', '临床表现']
['痞胀', '临床表现']
['早饱', '临床表现']
['嗳气', '临床表现']
['泛酸', '临床表现']
['嘈杂', '临床表现']
['恶心', '临床表现']
['呕吐', '临床表现']
['脉滑数', '临床表现']
['舌质淡', '临床表现']
['苔白灰（玉石状）', '临床表现']
['皮肤干燥', '临床表现']
['目光炯炯', '临床表现']
['舌苔厚腻', '临床表现']
['水肿', '中医诊断']
['气阴两虚', '中医证候']
['瘀血', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['慢性萎缩性胃炎', '西医诊断']
['腹泻', '临床表现']
['功能性消化不良', '西医诊断']
['匹维溴铵片', '西医治疗']
['腹泻型肠易激综合征', '西医诊断']
['功能性消化不良', '西医诊断']
['黄疸', '中医诊断']
['针刺', '中医治疗']
['柴半汤', '方剂']
['体表穴位电刺激', '中医治疗']
['肝胃不和', '中医证候']
['非糜烂性反流', '西医诊断']
['四联疗法', '其他治疗']
['中药足浴', '中医治疗']
['辨证针刺', '中医治疗']
['功能性消化不良

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 8 (char 7)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppDat

['儿童功能性消化不良', '西医诊断']
['腹针', '中医治疗']
['肠易激综合征', '西医诊断']
['非酒精性脂肪肝', '西医诊断']
['疏肝降脂片', '中医治疗']
['郁金', '中药']
['首乌', '中药']
['绵茵陈', '中药']
['枳实', '中药']
['山楂', '中药']
['川朴', '中药']
['丹参', '中药']
['功能性消化不良', '西医诊断']
['肝硬化腹水', '西医诊断']
['臌胀', '中医诊断']
['黄疸', '中医诊断']
['腹胀', '临床表现']
['口苦', '临床表现']
['功能性消化不良', '西医诊断']
['功能性便秘', '西医诊断']
['附子理中汤', '方剂']
['腹泻型肠易激综合征', '西医诊断']
['疏肝行气', '中医治则']
['调神解郁', '中医治则']
['推拿', '中医治疗']
['腹泻', '临床表现']
['寒热错杂', '中医证候']
['湿热雍盛', '中医证候']
['功能性消化不良', '西医诊断']
['西沙必利', '西医治疗']
['腹水', '临床表现']
['腹胀', '临床表现']
['纳差', '临床表现']
['乏力', '临床表现']
['调肝理脾通腑', '中医治则']
['便秘型肠易激综合征', '西医诊断']
['加味二神汤', '方剂']
['补骨脂', '中药']
['肉豆蔻', '中药']
['防风', '中药']
['白术', '中药']
['白术', '中药']
['二甲双胍', '西医治疗']
['非酒精性脂肪肝', '西医诊断']
['附片', '中药']
['干姜', '中药']
['猪苓', '中药']
['茯苓', '中药']
['高热', '临床表现']
['功能性消化不良', '西医诊断']
['上腹部疼痛', '临床表现']
['腹部胀气', '临床表现']
['嗳气', '临床表现']
['早饱', '临床表现']
['恶心', '临床表现']
['脾胃虚弱', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['功能性消化不良', '西医诊断']
['浮肿', '临床表现']
['卧床不起', '临床表现']
['失眠', '临床表现']
['心悸', '临床表现'

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 341, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 1 column 71 (char 70)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

['舌暗红', '临床表现']
['苔薄白', '临床表现']
['脉滑数', '临床表现']
['蛔厥', '中医诊断']
['清热利湿', '中医治则']
['疏肝健脾', '中医治则']
['解毒活血汤', '方剂']
['茵陈', '中药']
['大黄', '中药']
['山栀', '中药']
['虎杖', '中药']
['板蓝根', '中药']
['连翘', '中药']
['郁金', '中药']
['柴胡', '中药']
['当归', '中药']
['赤白芍', '中药']
['丹参', '中药']
['红花', '中药']
['枳壳', '中药']
['茯苓', '中药']
['甘草', '中药']
['便秘', '中医诊断']
['腹痛', '临床表现']
['拔罐', '中医治疗']
['胃食管反流性咳嗽', '西医诊断']
['茵陈', '中药']
['神阙隔物灸', '中医治疗']
['针刺', '中医治疗']
['腹泻型肠易激综合征', '西医诊断']
['浮肿', '临床表现']
['心烦', '临床表现']
['口苦', '临床表现']
['口渴', '临床表现']
['不欲饮', '临床表现']
['小便黄赤', '临床表现']
['苔黄腻', '临床表现']
['肝郁脾虚', '中医证候']
['脾胃病', '西医诊断']
['肝经湿热蕴结', '中医证候']
['电针', '中医治疗']
['健脾祛湿方', '方剂']
['非酒精性脂肪肝', '西医诊断']
['功能性消化不良', '西医诊断']
['中医食疗', '中医治疗']
['黄连素', '西医治疗']
['辨证针刺', '中医治疗']
['便秘', '中医诊断']
['疏肝健脾', '中医治则']
['非酒精性脂肪肝', '西医诊断']
['舌苔白厚腻', '临床表现']
['脉沉伏', '临床表现']
['水样便', '临床表现']
['功能性便秘', '西医诊断']
['气阴两虚', '中医证候']
['气短乏力', '临床表现']
['心悸', '临床表现']
['口干', '临床表现']
['排便无力', '临床表现']
['补气滋阴通腑汤', '方剂']
['面色晦暗', '临床表现']
['形体消瘦', '临床表现']
['腹大如瓮，鼓之如鼓', '临床表现']

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 8 (char 7)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppDat

['针刺', '中医治疗']
['耳压', '中医治疗']
['面色苍白', '临床表现']
['少气懒言', '临床表现']
['贫血貌', '临床表现']
['面色少泽', '临床表现']
['白睛轻度黄染', '临床表现']
['苔白厚腻', '临床表现']
['耳穴贴压', '中医治疗']
['心悸', '中医诊断']
['瘀阻心脉', '中医证候']
['麻仁软胶囊', '中医治疗']
['热积证功能性便秘', '西医诊断']
['茵陈', '中药']
['黄芩', '中药']
['枳壳', '中药']
['姜黄', '中药']
['大黄', '中药']
['茯苓', '中药']
['茅根', '中药']
['椒目', '中药']
['银花', '中药']
['芦根', '中药']
['生地', '中药']
['四逆散', '方剂']
['肝胃郁热', '中医证候']
['食疗', '其他治疗']
['吗丁啉', '西医治疗']
['小儿功能性消化不良', '西医诊断']
['吗丁啉', '西医治疗']
['指针', '中医治疗']
['竹药罐', '中医治疗']
['脾胃虚弱', '中医证候']
['功能性消化不良', '西医诊断']
['发热', '临床表现']
['畏寒', '临床表现']
['寒战', '临床表现']
['消瘦', '临床表现']
['苔薄黄', '临床表现']
['枸杞', '中药']
['脾阳虚', '中医证候']
['大腹胀', '临床表现']
['齿衄', '临床表现']
['郁金', '中药']
['枳壳', '中药']
['川木香', '中药']
['功能性消化不良', '西医诊断']
['利湿', '中医治则']
['清热解毒', '中医治则']
['退黄', '中医治则']
['耳穴贴压', '中医治疗']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['功能性消化不良', '西医诊断']
['气阴两虚', '中医证候']
['芪榔方', '方剂']
['温针灸', '中医治疗']
['慢性萎缩性胃炎', '西医诊断']
['脘腹胀满', '临床表现']
['胃脘隐痛', '临床表现']
['纳差食少', '临床表现']
['嗳气呃逆', '临床表现']
['功能性消化不良', '

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 356, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 3 (char 2)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ip

['慢性萎缩性胃炎', '西医诊断']
['气阴不足', '中医证候']
['麻枳降浊方', '方剂']
['心悸', '临床表现']
['乏力', '临床表现']
['肝硬化腹水', '西医诊断']
['功能性消化不良', '西医诊断']
['四磨汤', '方剂']
['肝脾不和', '中医证候']
['香砂六君子汤加味', '方剂']
['木香', '中药']
['九香虫', '中药']
['砂仁', '中药']
['党参', '中药']
['炒白术', '中药']
['姜半夏', '中药']
['陈皮', '中药']
['脾胃虚寒', '中医证候']
['功能性消化不良', '西医诊断']
['中虚外寒侵袭', '中医证候']
['柴胡疏肝散', '方剂']
['萎缩性胃炎', '西医诊断']
['脾胃虚弱', '中医证候']
['双歧三联活菌', '西医治疗']
['成人功能性便秘', '西医诊断']
['腹水', '西医诊断']
['善胃Ⅰ号方', '方剂']
['血瘀热毒', '中医证候']
['胃癌前病变', '西医诊断']
['脾胃虚寒', '中医证候']
['肝郁气滞', '中医证候']
['脾胃湿热', '中医证候']
['寒热错杂', '中医证候']
['寒热错杂', '中医证候']
['慢性萎缩性胃炎', '西医诊断']
['半夏泻心汤加味', '方剂']
['柴平舒胃汤加减', '方剂']
['功能性消化不良', '西医诊断']
['柴胡疏肝散', '方剂']
['功能性消化不良', '西医诊断']
['心肺阴性', '临床表现']
['腹胀', '临床表现']
['舌质红', '临床表现']
['苔薄黄', '临床表现']
['脉弦数有力', '临床表现']
['[["慢性萎缩性胃炎", "西医诊断"], ["气阴不足", "中医证候"]]', '[["麻枳降浊方", "方剂"]]', '[["心悸", "临床表现"]]', '[["乏力", "临床表现"], ["肝硬化腹水", "西医诊断"]]', '[["功能性消化不良", "西医诊断"], ["四磨汤", "方剂"], ["肝脾不和", "中医证候"]]', '[["香砂六君子汤加味", "方剂"], ["木香", "中药"], ["九香虫", "中药"], ["砂仁", "中药"

Traceback (most recent call last):
  File "C:\Users\lkjx0\AppData\Local\Temp\ipykernel_31488\2331773926.py", line 25, in predict_collect
    output = json.loads(output.strip().replace('\n', '').replace('“', '"').replace('”', '"'))  # 去除多余的换行符和空格
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 338, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\software\anaconda\envs\pytorch\Lib\json\decoder.py", line 354, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 11 (char 10)
Traceback (most recent call last):
  File "C:\Users\lkjx0\AppD

In [30]:
len(set_a_strict),  len(set_b_strict)

(38, 35)

In [27]:
f1_score(set_a_strict, set_b_strict), f1_score(set_a_relax, set_b_relax)

(0.45525902668759816, 0)

## 单样本推断

In [ ]:
import tqdm
system_prompt = "你是一个中医药领域的命名体识别专家，你需要从给定句子中提取相关命名体，以json格式输出，例如：[[', 4, '口苦', '临床表现'],[5, 6, '口干', '临床表现']]，其中每个元素分别表示命名体的起始位置、结束位置、文本内容和标签类型。如果没有则返回空列表[]"

set_a_strict = set()
set_b_strict = set()
set_a_relax = set()
set_b_relax = set()

pbar = tqdm.tqdm(total=len(test_data.data), desc="Evaluating on test set")

for i in test_data.data:
    messages = [
        {"role": "system", "content": system_prompt},
        {"role": "user", "content": f"{i['input']}"},
    ]
    inputs = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True) # 输出会停在最后一个user，然后添加assistant的开始标记等待模型生成, 不会编码assistant的内容
    inputs = tokenizer([inputs], return_tensors="pt").to(peft_model.device)
    with torch.no_grad():
        outputs = peft_model.generate(inputs.input_ids, max_new_tokens=512)
    
    output = outputs[0][len(inputs.input_ids[0]):]  # 获取生成的部分
    decoded_output = tokenizer.decode(output, skip_special_tokens=True)
    decoded_output = json.loads(decoded_output.strip().replace('\n', ''))  # 去除多余的换行符和空格
    print(decoded_output)
    print(i['output'])


    for item in json.loads(i['output']):
        set_a_strict.add(tuple(item))
        set_a_relax.add((item[2], item[3]))  # 仅比较实体内容和标签类型，忽略位置

    try:
        for item in decoded_output:
            set_b_strict.add(tuple(item))
            set_b_relax.add((item[2], item[3]))  # 仅比较实体内容和标签类型，忽略位置
    except:
        import traceback
        print(i['output'])
        print(decoded_output)
        traceback.print_exc()
    finally:
        pbar.update(1)


In [78]:
del peft_model

In [79]:
import gc
gc.collect()

51734

In [80]:
torch.cuda.empty_cache()